src/Mir/Trans.hs

```diff
@@ -523,7 +528,9 @@ transNullaryOp M.Box ty = do
         vals <- mapM (\f -> mkBox $ f ^. fty) (v ^. vfields)
         buildStruct adt vals
     mkBox ty = mirFail $ "unsupported type in mkBox: " ++ show ty
-transNullaryOp _ _ = mirFail "nullop"
+transNullaryOp M.SizeOf _ = do
+    -- TODO: return the actual size, once mir-json exports size/layout info
```

Comment:
Exporting just size/alignment would be easy.  Layout would be a bit more work, since rustc supports some tricky layout optimizations that we'd need to handle.

In the future, we'll have to export full layouts as part of expanded unsafe code support, at which point implementing this operation will be trivial.  But until we have that unsafe support, I don't think user code can do much that's interesting with `size_of` anyway.